### PR TITLE
Use an explicit server image tag when provided

### PR DESCRIPTION
**Pull request by tdeebswihart**
## What was changed

When running against a provided docker image we now use the _explicit_ image tag I've embedded in our artifacts rather than `latest`.

## Why?

While adding feature tests to our [builds repo](https://github.com/temporalio/docker-builds/pull/200) I discovered that running `latest` doesn't always run the actual image we've loaded. It seems to be running the latest version we've pushed to dockerhub rather than the locally tagged `latest` version.

If this is true (it's what my testing indicates), then we aren't testing the image we think we are in the server's CI. This will fix that, however

### CODE REVIEW

File: dockerfiles/docker-compose.for-server-image.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version: '3.5'
 
 services:
   temporal-server:
-    image: temporaliotest/auto-setup:latest
+    image: temporaliotest/auto-setup:${IMAGE_TAG:-latest}
```
Comment (Sushisource) [inline dockerfiles/docker-compose.for-server-image.yaml:7]: Where does the dash come from?

**Review by Sushisource — APPROVED** [review]
Makes sense to me

File: dockerfiles/docker-compose.for-server-image.yaml (hunk 1 above)
Comment (tdeebswihart) [reply dockerfiles/docker-compose.for-server-image.yaml:7]: https://docs.docker.com/compose/environment-variables/env-file/#interpolation

**Comment by tdeebswihart** [comment]
Hrm, it looks like `uses` won't work like this when called cross-repo: https://github.com/temporalio/docker-builds/actions/runs/9177587129/job/25235699370?pr=200

That's obnoxious

File: dockerfiles/docker-compose.for-server-image.yaml (hunk 1 above)
Comment (Sushisource) [reply dockerfiles/docker-compose.for-server-image.yaml:7]: Ah, got it. That syntax is, well, it's a syntax.

**Comment by Sushisource** [comment]
> Hrm, it looks like `uses` won't work like this when called cross-repo: https://github.com/temporalio/docker-builds/actions/runs/9177587129/job/25235699370?pr=200
> 
> That's obnoxious

Yah I want to say that's why I didn't do it that way in the first place. GH Actions has some very weird limitations.

**Comment by tdeebswihart** [comment]
> > Hrm, it looks like `uses` won't work like this when called cross-repo: https://github.com/temporalio/docker-builds/actions/runs/9177587129/job/25235699370?pr=200
> > That's obnoxious
> 
> Yah I want to say that's why I didn't do it that way in the first place. GH Actions has some very weird limitations.

Yeah. I've ripped it out with my latest commit. That's deeply unfortunate

Maybe we should rewrite our workflows in Temporal :P

**Comment by tdeebswihart** [comment]
The final action passed (see https://github.com/temporalio/features/actions/runs/9178336653/job/25238013805?pr=484), the above is out of date.

**Pull request merged**